data/scripts/lib/register_monster_type.lua

```diff
@@ -259,127 +259,120 @@ registerMonsterType.immunities = function(mtype, mask)
-					end
-					if attack.shootEffect then
-						spell:setCombatShootEffect(attack.shootEffect)
-					end
-					if attack.name == "drunk" then
-						spell:setConditionType(CONDITION_DRUNK)
-						if attack.drunkenness then
-							spell:setConditionDrunkenness(attack.drunkenness)
-						end
+function AbilityTableToSpell(ability)
+	local spell = MonsterSpell()
+	if ability.name then
+		if ability.name == "melee" then
+			spell:setType("melee")
+			if ability.attack and ability.skill then
+				spell:setAttackValue(ability.attack, ability.skill)
+			end
+			if ability.minDamage and ability.maxDamage then
+				spell:setCombatValue(ability.minDamage, ability.maxDamage)
+			end
+			if ability.interval then
+				spell:setInterval(ability.interval)
+			end
+			if ability.effect then
+				spell:setCombatEffect(ability.effect)
+			end
+		else
+			spell:setType(ability.name)
+			if ability.type then
+				spell:setCombatType(ability.type)
+			end
+			if ability.interval then
+				spell:setInterval(ability.interval)
+			end
+			if ability.chance then
+				spell:setChance(ability.chance)
+			end
+			if ability.range then
+				spell:setRange(ability.range)
+			end
+			if ability.duration then
+				spell:setConditionDuration(ability.duration)
+			end
+			if ability.speed then
+				if type(ability.speed) ~= "table" then
+					spell:setConditionSpeedChange(ability.speed)
+				elseif type(ability.speed) == "table" then
+					if ability.speed.min and ability.speed.max then
+						spell:setConditionSpeedChange(ability.speed.min, ability.speed.max)
 					end
 				end
-			elseif attack.script then
-				spell:setScriptName(attack.script)
-				if attack.interval then
-					spell:setInterval(attack.interval)
-				end
-				if attack.chance then
-					spell:setChance(attack.chance)
-				end
-				if attack.minDamage and attack.maxDamage then
-					spell:setCombatValue(attack.minDamage, attack.maxDamage)
-				end
-				if attack.target then
-					spell:setNeedTarget(attack.target)
-				end
-				if attack.direction then
-					spell:setNeedDirection(attack.direction)
+			end
+			if ability.target then
+				spell:setNeedTarget(ability.target)
+			end
+			if ability.length then
+				spell:setCombatLength(ability.length)
+			end
+			if ability.spread then
+				spell:setCombatSpread(ability.spread)
+			end
+			if ability.radius then
+				spell:setCombatRadius(ability.radius)
+			end
+			if ability.minDamage and ability.maxDamage then
+				spell:setCombatValue(ability.minDamage, ability.maxDamage)
+			end
+			if ability.effect then
+				spell:setCombatEffect(ability.effect)
+			end
+			if ability.shootEffect then
+				spell:setCombatShootEffect(ability.shootEffect)
+			end
+			if ability.name == "drunk" then
+				spell:setConditionType(CONDITION_DRUNK)
+				if ability.drunkenness then
+					spell:setConditionDrunkenness(ability.drunkenness)
 				end
 			end
+		end
+		if ability.condition then
+			if ability.condition.type then
+				print(ability.condition.type)
```

Comment:
and this?

Reply:
and why not make the `AbilityTableToSpell` function local? from what I see, it will only be used in this file